jobs/release.Jenkinsfile

```diff
@@ -96,11 +99,16 @@ podTemplate(cloud: 'openshift', label: pod_label, yaml: pod) {
             """)
         }
 
-        def basearches = ""
-        if (params.ARCHES == '') {
-            basearches = shwrapCapture("jq -r '.builds | map(select(.id == \"${params.VERSION}\"))[].arches[]' builds/builds.json").split()
-        } else {
-            basearches = params.ARCHES.split()
+        def basearches = params.ARCHES.split()
+        def builtarches = shwrapCapture("jq -r '.builds | map(select(.id == \"${params.VERSION}\"))[].arches[]' builds/builds.json").split()
+        if builtarches.toSet() != basearches.toSet() {
+            if params.ALLOW_MISSING_ARCHES {
+                echo "Some requested architectures did not successfully build! Continuing."
+                basearches = builtarches
```

Comment:
And then here

```suggestion
                basearches = builtarches.intersect(basearches)
```

Reply:
And also maybe an `assert` here to sanity-check that `x86_64` is there.

Reply:
I think we should keep the basearches in the parameter as the source of truth. I.e. we shouldn't release for an arch just because it's built. So then all `ALLOW_MISSING_ARCHES` does is to allow releasing a subset, not a different set.

Reply:
ok I think I get it now. I needed to work through an example:

```
builtarches = ["1", "2", "3"]
basearches = ["1", "2", "4"]

basearches = builtarches.intersect(basearches) -> ["1", "2"]
```

Wheras my code would have `basearches` as `["1", "2", "3"]`